lib/services/cordova-migration-service.ts

```diff
@@ -134,6 +143,153 @@ export class CordovaMigrationService implements ICordovaMigrationService {
+			});
+			this.invalidPlugins = _.filter(allMarketplacePlugins, plugin => {
+				let pluginBasicInformation = this.$pluginsService.getPluginBasicInformation(plugin);
+				return !this.$pluginsService.isPluginSupported(pluginBasicInformation.name, pluginBasicInformation.version, newVersion);
+			});
+
+			if (this.invalidPlugins.length > 0) {
+				this.promptUserForInvalidPluginsAction(this.invalidPlugins, newVersion).wait();
+			}
+
+			_.each(this.$project.configurations, (configuration: string) => {
+				let oldPluginsList = this.$project.getProperty("CorePlugins", configuration);
+				let newPluginsList = this.migratePlugins(oldPluginsList, oldVersion, newVersion).wait();
+				this.$logger.trace("Migrated core plugins to: ", helpers.formatListOfNames(newPluginsList, "and"));
+				this.$project.setProperty("CorePlugins", newPluginsList, configuration);
+			});
+
+			let backedUpFiles: string[] = [],
+				backupSuffix = ".backup";
+			try {
+				_.each(this.$mobileHelper.platformNames, (platform) => {
+					this.$logger.trace("Replacing cordova.js file for %s platform ", platform);
+					let cordovaJsFileName = path.join(this.$project.getProjectDir().wait(), `cordova.${platform}.js`.toLowerCase());
+					let cordovaJsSourceFilePath = this.$resources.buildCordovaJsFilePath(newVersion, platform);
+					this.$fs.copyFile(cordovaJsFileName, cordovaJsFileName + backupSuffix).wait();
+					backedUpFiles.push(cordovaJsFileName);
+					this.$fs.copyFile(cordovaJsSourceFilePath, cordovaJsFileName).wait();
+				});
+			} catch(error) {
+				_.each(backedUpFiles, file => {
+					this.$logger.trace("Reverting %s", file);
+					this.$fs.copyFile(file + backupSuffix, file).wait();
+				});
+				this.$errors.failWithoutHelp(error.message);
+			}
+			finally {
+				_.each(backedUpFiles, file => {
+					this.$fs.deleteFile(file + backupSuffix).wait();
+				});
+			}
+
+			this.$logger.info("Successfully migrated to version %s", versionDisplayName);
+		}).future<void>()();
+	}
+
+	public getSupportedPlugins(): IFuture<string[]> {
+		return (() => {
+			let version: string;
+			if(this.$project.projectData) {
+				version = this.$project.projectData.FrameworkVersion;
+			} else {
+				let selectedFramework = _.findLast(this.getSupportedFrameworks().wait(), (sv: Server.FrameworkVersion) => sv.DisplayName.indexOf(this.$projectConstants.EXPERIMENTAL_TAG) === -1);
+				version = selectedFramework.Version;
+			}
+
+			return this.pluginsForVersion(version).wait();
+		}).future<string[]>()();
+	}
+
+	private promptUserForInvalidPluginsAction(plugins: string[], toVersion: string): IFuture<Boolean> {
+		return (() => {
+			let remove = `Remove those plugins from all configurations`;
+			let cancel = 'Cancel Cordova migration';
+			let choice = this.$prompter.promptForChoice(`The plugins ${plugins.join(', ')} are not supported for Cordova version ${toVersion} what do you want to do?`, [remove, cancel]).wait();
```

Comment:
in case there's only one plugin, that is not supported, the questions is:
```
? The plugins com.telerik.plugins.nativepagetransitions@0.3.0 are not supported for Cordova version 3.0.0 what do you want to do? (Use arrow keys)
```

Also do not hardcode Cordova, use this.$project.projectData.Framework instead :smile_cat: